src/interchange/src/json.rs

```diff
@@ -377,16 +381,32 @@ fn build_row_schema_fields(
     columns: &[(ColumnName, ColumnType)],
     type_namer: &mut Namer,
     custom_names: &BTreeMap<GlobalId, String>,
+    set_null_defaults: bool,
 ) -> Vec<serde_json::Value> {
     let mut fields = Vec::new();
     let mut field_namer = Namer::default();
     for (name, typ) in columns.iter() {
         let (name, _seen) = field_namer.valid_name(name.as_str());
-        let field_type = build_row_schema_field(type_namer, custom_names, typ);
-        fields.push(json!({
-            "name": name,
-            "type": field_type,
-        }));
+        let field_type =
+            build_row_schema_field_type(type_namer, custom_names, typ, set_null_defaults);
+
+        // It's a nullable union if the type is an array and the first option is "null"
+        let is_nullable_union = field_type
+            .as_array()
+            .is_some_and(|array| array.first().is_some_and(|first| first == &json!("null")));
```

Comment:
@umanwizard Updated the logic here

Reply:
Looks good to me. Let's also add a comment in `build_row_schema_field_type` that we need to refactor things if we ever want to support different kinds of union schema (and that `"null"` must always come first in the union for nullable types)

Reply:
done!